### DISCUSSION TITLE
Verse Number Formatting

### DISCUSSION THREAD

**Discussion by mattlynskey** [tim-hub/obsidian-bible-reference#69] [Ideas]
Thanks so much for this great app! I like the addition of various verse numbering formats. Is it possible to add a verse numbering format with just the verse number and space (no punctuation) **as it appears in a typical Bible translation**? Thanks again for your great work on this!

**Comment by mattlynskey** [comment]
Also, is there any way to add one space (on either side of the verse number) between the verse numbers and the text? See the spacing in the image below. 

Again, thanks for your great work on this!

<img width="528" alt="Screen Shot 2022-12-23 at 3 20 12 PM" src="https://user-images.githubusercontent.com/74412970/209402282-6ae2f8a0-6f20-43c6-ba92-abcab8e0e17a.png">

**Reply by JPT62089** [reply]
I'm seeing that too... Thought I had that resolved. I unfortunately won't have time to take a look at it in the next week but perhaps if @tim-hub hasn't yet I'll take a stab next weekend or early Jan. Should be an easy fix tho :)

**Reply by mattlynskey** [reply]
Thanks for the reply @JPT62089! Appreciate the help on this!

**Reply by KINGDOMMAN** [reply]
I think I probably need a step-by-step tutorial on how to make this plugin work. I have installed, enabled it, and the gif seems to tell you to type the verses in the prescribed syntax (which I am doing), but nothing shows up. What could I not be doing right? Someone help, please.

**Reply by JPT62089** [reply]
Hey @KINGDOMMAN, can you share how you're typing it? Also what kind of device are you using? I know there are some reports of it not working on a bunch of mobile device (Tablets and phones mostly).

In the meantime make sure you're typing verse references with no spaces. Here's a few examples.

--John1:1-5
--1Cor1:6-10

Also make sure you type it correctly the first time. I've noticed sometimes it doesn't always trigger properly when spaces are added and then removed. Currently my only experience is on Windows so your mileage may vary depending on what device you have.

**Comment by JPT62089** [comment]
@mattlynskey 

Verse Number only and the spacing issue should be resolved in PR #71. @tim-hub will be the one able to review and make any refactoring to my slop ( ;P ) and potentially push out those changes.

Let me know if you noticed any further issues in the meantime.

**Reply by tim-hub** [reply]
Thanks, will try to review it today

**Comment by mattlynskey** [comment]
Thanks for the help and work on this @JPT62089 and @tim-hub.

Another spacing/formatting issue that I notice is the lack of a space for books of the Bible with numbers. For instance when I type in --1John2:2, the rendered verse in the output becomes 1John 2:2 (there is a needed space between "1" and "J"). See images below

Again, this is a great plugin that is proving useful to me. Thanks for the effort to make and refine it!
<img width="585" alt="Screen Shot 2023-01-09 at 10 37 20 AM" src="https://user-images.githubusercontent.com/74412970/211346920-2c553146-b16c-41ba-880a-a881c5c590f3.png">
<img width="527" alt="Screen Shot 2023-01-09 at 10 37 53 AM" src="https://user-images.githubusercontent.com/74412970/211346926-4ba9e500-ffa9-4bb8-9feb-53942df37692.png">

**Reply by JPT62089** [reply]
Hey @mattlynskey

I've noticed this too. I know exactly what the issue is and looked into a simple resolution with my last updates I did tho it will be a bit more complicated than I had hoped. Next time I have some time to tackle some issues and features I'll take a look at this if it's still an issue.

Basically it's pulling the exact text that you type in for the verse reference. I'm hoping to work through a way to get a correctly formatted version. Tho this will be tricky since there's multiple languages at play here. I believe it can be done via the APIs but have yet to look into it.
